api/client.go

```diff
@@ -12,6 +12,7 @@ import (
 	"strings"
 
 	genq "github.com/Khan/genqlient/graphql"
+	"github.com/superfly/flyctl/internal/instrument"
```

Comment:
In Go, packages (things you declare with `package xxx`) and modules (things on go.mod) are different. Module-wise, `github.com/superfly/flyctl/api` depends `github.com/superfly/flyctl` but `flyctl` also depends `api`.

If flyctl mostly calls RunWithContext() and a few methods, you can define a wrapper struct like https://github.com/containerd/containerd/blob/6020903f2c2f0b6b148b02ca7bdd1cd9c6efac3d/pkg/cri/instrument/instrumented_service.go#L50 to call GraphQL.Begin(). Another option may be defining an interface for instrumentation and pass its implementation, which may be something similar to https://github.com/open-telemetry/opentelemetry-go.

Reply:
Oh interesting. It looks like `api` is a special case here being mentioned in `go.mod`?